compiler/src/dmd/semantic2.d

```diff
@@ -275,6 +275,7 @@ private extern(C++) final class Semantic2Visitor : Visitor
             // https://issues.dlang.org/show_bug.cgi?id=20417
             // Don't run CTFE for the temporary variables inside typeof or __traits(compiles)
             vd._init = vd._init.initializerSemantic(sc, vd.type, sc.intypeof == 1 || sc.flags & SCOPE.compile ? INITnointerpret : INITinterpret);
+            tryLowerStaticAA(vd, sc);
```

Comment:
So it turns out classes get initializerSemantic run twice, so this does the lowering a second time, which is a bit wasteful. Perhaps the `VarDeclaration` / `Initializer` should get a `semanticDone` flag.

Reply:
VarDeclaration has a semanticRun field. Doesn't that help?